src/lib.rs

```diff
@@ -602,6 +602,19 @@ where
         sequence_number: u32,
     ) -> Result<SecurityStatus>;
 
+    /// TODO
+    fn make_signature(&mut self,
+        flags: SignatureFlags,
+        message: &mut [SecurityBuffer],
+        sequence_number: u32,
+    ) -> crate::Result<SecurityStatus>;
```

Comment:
```suggestion
    ) -> crate::Result<()>;
```

You should not use `SecurityStatus` as a return value. The `SecurityStatus` should be used only for informing the caller about the current status of the authentication. See the following table: https://learn.microsoft.com/en-us/windows/win32/secauthn/initializesecuritycontext--kerberos#return-value.
According to the documentation (https://learn.microsoft.com/en-us/windows/win32/api/sspi/nf-sspi-makesignature), the `MakeSignature` function returns the error code (including `SEC_E_OK`). So, `Result<()>` will be enough.